### PR TITLE
T1016 System network configuration discovery

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature
> Implements T1016 System network configuration discovery attack technique
image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js [hunk 1]
```diff
@@ -11,7 +11,46 @@ export function renderMachine(val){
...
+        {Header: 'Usage',
+          id: 'usage',
+          accessor: x => x._id.usage,
+          style: { 'whiteSpace': 'unset' }}]
+    }])}
+
+/* Renders fields that contains 'used' boolean value and 'name' string value.
+'Used' value determines if 'name' value will be shown.
+ */
+export function renderCollections(info){
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js]: maybe rename to renderUsedCollections?

File: monkey/monkey_island/cc/services/attack/technique_reports/T1016.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    query = [{'$match': {'telem_category': 'system_info'}},
+             {'$project': {'machine': {'hostname': '$data.hostname', 'ips': '$data.network_info.networks'},
+                           'networks': '$data.network_info.networks',
+                           'netstat': '$data.network_info.netstat'}},
+             {'$addFields': {'_id': 0,
+                             'netstat': 0,
+                             'networks': 0,
+                             'info': [
+                                 {'used': {'$and': [{'$ifNull': ['$netstat', False]}, {'$gt': ['$netstat', {}]}]},
+                                  'name': {'$literal': 'Network connections (via netstat command)'}},
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1016.py]: Just `'Network connections (netstat)'` is enough

File: monkey/monkey_island/cc/services/attack/technique_reports/T1016.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                                 {'used': {'$and': [{'$ifNull': ['$networks', False]}, {'$gt': ['$networks', {}]}]},
+                                  'name': {'$literal': 'Network interface info'}},
+                             ]}}]
+
+    @staticmethod
+    def get_report_data():
+        network_info = list(mongo.db.telemetry.aggregate(T1016.query))
+        if network_info:
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1016.py]: status = USED if network_info else UNSCANNED

File: monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js (hunk 1 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js]: You are right, naming here is terrible

**Pull request merged**